src/main/resources/static/js/dm/viewer/TextEditor.js

```diff
@@ -172,7 +172,11 @@ dm.viewer.TextEditor.prototype.saveContents = function() {
     var highlightPlugin = this.field.getPluginByClassId('Annotation');
     highlightPlugin.updateAllHighlightResources();
 
-    this.databroker.sync();
+    var request = this.databroker.sync();
+    request.fail(function(response) {
+      this.saveError = true;
+      alert("Error: failed to save changes to text resource. Please refresh your browser window to avoid losing progress.");
```

Comment:
Would this lead to people reloading the page with unsaved data, and therefore lose that data?

Reply:
Good points @jamiefolsom  — I was envisioning a use case where a problem with the user's session necessitates reconnecting to the backend (vs. the test case of breaking that link via device internet connection), but the UX should also account for device connection problems. If it's a session problem, a refresh would be needed since the user would need to log back in. What do you think of "Error: failed to save changes to text resource. Please copy any in-progress text to a local file, then check your internet connection and refresh your browser window."?